Rna seqdecider

wshands opened the pull request:
Brian, please take a look at this pull request for RNASeq script. It should be just an additional file RNASeqTask.py 

I tested it by making a call to Luigi.
PYTHONPATH='..' luigi --module RNASeqTask RNASeqCoordinator --redwood-client-path ../../../redwood-client/ucsc-storage-client/ --redwood-host storage.ucsc-cgl.org --redwood-token --local-scheduler --tmp-dir pwd/luigi_state --data-dir $(pwd) --max-jobs 400 1> stdout.txt 2> stderr.txt

The problem I see is that the script seems to run twice. I still have to debug this. There are a couple of other points:

I had to instruct urlopen to ignore certs due to the non self signed certs issue
I do not know where we will have rsem, kallisto and star so I default them to the cwd
The data file location setup in the json around line 71 will need to be changed

Commit:
Initial checkin for the RNA-Seq decider

Initial checkin for the RNA-Seq decider

Commit:
Fixed initial checkin for RNA-Seq decider

Fixed initial checkin for RNA-Seq decider

rcurrie commented:
I put the rnaseq 2.0.8 reference files on the internet when we setup BC to run things:

https://github.com/UCSC-Treehouse/pipelines/blob/master/rnaseq-cgl-pipeline/Makefile

I also have them in ceph and azure so that when we spin up dockers in treehouse it pulls from the closest location. If relevant might make sense to have it see if they are there and if not pull them. I think we’ll want to have reference files in each environments object store so vms can pull them quickly and with no internet bandwidth.

Rob
> On Dec 7, 2016, at 10:22 PM, Walter Shands <notifications@github.com> wrote:
> 
> Brian, please take a look at this pull request for RNASeq script. It should be just an additional file RNASeqTask.py
> 
> I tested it by making a call to Luigi.
> PYTHONPATH='..' luigi --module RNASeqTask RNASeqCoordinator --redwood-client-path ../../../redwood-client/ucsc-storage-client/ --redwood-host storage.ucsc-cgl.org --redwood-token --local-scheduler --tmp-dir pwd/luigi_state --data-dir $(pwd) --max-jobs 400 1> stdout.txt 2> stderr.txt
> 
> The problem I see is that the script seems to run twice. I still have to debug this. There are a couple of other points:
> 
> I had to instruct urlopen to ignore certs due to the non self signed certs issue
> I do not know where we will have rsem, kallisto and star so I default them to the cwd
> The data file location setup in the json around line 71 will need to be changed
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/BD2KGenomics/dcc-action-service/pull/2
> 
> Commit Summary
> 
> 	• Initial checkin for the RNA-Seq decider
> 	• Fixed initial checkin for RNA-Seq decider
> File Changes
> 
> 	• A luigi_task_executor/RNASeqTask.py (329)
> Patch Links:
> 
> 	• https://github.com/BD2KGenomics/dcc-action-service/pull/2.patch
> 	• https://github.com/BD2KGenomics/dcc-action-service/pull/2.diff
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

Commit:
Added unique id for Task

Added unique id for ConsonanceTask and fixed method calls
so that Coordinator completed properly

briandoconnor commented:
Walt, you dropped off the rsem etc ref files on S3 already, I turned these paths into public URLs.  I point to these locations in my sample JSON config I left on the Consonance VM in us-west-2, see that for an example.  It totally makes sense to upload these to redwood and just use the same URL syntax as the input files to retrieve them.
